lib/dereference.ts

```diff
@@ -123,7 +123,27 @@ function crawl<S extends object = JSONSchema, O extends ParserOptions<S> = Parse
             circular = dereferenced.circular;
             // Avoid pointless mutations; breaks frozen objects to no profit
             if (obj[key] !== dereferenced.value) {
+              // If we have properties we want to preserve from our derefernced schema then we need
+              // to copy them over to our new object.
+              const preserved: Map<string, unknown> = new Map();
```

Comment:
nit: can we wrap this logic and this object allocation inside a check for if `derefOptions?.preservedProperties` is set? if its not set we can skip a lot of this logic.

Reply:
Absolutely!